test/decorator.test.js

```diff
@@ -104,58 +112,69 @@ test('should pass error for missing request decorator', t => {
   fastify
     .register(plugin)
     .ready((err) => {
-      t.type(err, Error)
-      t.match(err, /The decorator 'foo'/)
+      t.assert.ok(err instanceof Error)
+      t.assert.ok(err.message.includes("'foo'"))
+      done()
     })
 })
 
-test('decorateReply inside register', t => {
+test('decorateReply inside register', (t, done) => {
   t.plan(11)
   const fastify = Fastify()
 
   fastify.register((instance, opts, done) => {
     instance.decorateReply('test', 'test')
 
     instance.get('/yes', (req, reply) => {
-      t.ok(reply.test, 'test exists')
+      t.assert.ok(reply.test, 'test exists')
       reply.send({ hello: 'world' })
     })
 
     done()
   })
 
   fastify.get('/no', (req, reply) => {
-    t.notOk(reply.test)
+    t.assert.ok(!reply.test)
     reply.send({ hello: 'world' })
   })
 
   fastify.listen({ port: 0 }, err => {
-    t.error(err)
-    t.teardown(() => { fastify.close() })
+    t.assert.ifError(err)
+    t.after(() => fastify.close())
+
+    let pending = 2
+
+    function completed () {
+      if (--pending === 0) {
+        done()
+      }
+    }
```

Comment:
Shouldn't the plan be enough? Why do we need the `done` callback when using a plan?

Reply:
I had put the `completed()` function, because by having two `sget`, the last `sget` could end before the other and the test would still be completed.

Reply:
🤔 I'll have to research this. Maybe core needs tspl's `await plan`.

attn: @mcollina

Reply:
If we want to "correct" this without the utility function, and without core being fixed, we'd move the second request into the callback of the first. Probably.

Reply:
I opened https://github.com/nodejs/node/issues/56758

Reply:
I would not hold this PR until the new feature is being shipped.

If we don't like the code, we may use an `hapi/teamwork`-like utility:
https://github.com/hapijs/teamwork/blob/baadcc8bc0800ed74d0b57fa468983bf6c1fc9e9/test/index.js#L29-L46

I use it often

Reply:
Can we merge this PR ?